src/main/java/org/broadinstitute/consent/http/service/DacService.java

```
@@ -179,9 +183,17 @@ public void updateDac(String name, String description, String email, Integer dac
     dacDAO.updateDac(name, description, email, updateDate, dacId);
   }
 
-  public void deleteDac(Integer dacId) {
-    dacDAO.deleteDacMembers(dacId);
-    dacDAO.deleteDac(dacId);
+  public void deleteDac(Integer dacId) throws IllegalArgumentException, SQLException {
+    Dac fullDac = dacDAO.findById(dacId);
+    // TODO: Broad DAC logic will be updated with DCJ-498 to not be reliant on name
+    if (fullDac.getName().toLowerCase().contains("broad")) {
+      throw new IllegalArgumentException("This is the Broad DAC, which can not be deleted.");
+    }
+    try {
+      dacServiceDAO.deleteDacAndDaas(fullDac);
+    } catch (IllegalArgumentException e) {
```

I see we swallow the original exception in full, is there any value in logging it for debugging even if we don't propagate it to the user?

---

Yes, there is some value to that, thank you